modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/cache/ClientCachePartitionsRequest.java

```diff
@@ -95,79 +92,43 @@ public ClientCachePartitionsRequest(BinaryRawReader reader) {
             if (!cacheDesc.cacheType().userCache())
                 continue;
 
-            processCache(ctx, groups, cacheGroupIds, affinityVer, cacheDesc);
+            ClientCachePartitionAwarenessGroup grp = processCache(ctx, affinityVer, cacheDesc);
+
+            if (grp == null)
+                continue;
+
+            if (grps.containsKey(grp))
```

Comment:
Lets use something like
```
grp0 = grps.get(grp);
if (grp0 != null)
  grp.addCache(cacheDesc)
```
`grp.equals` and `grp.hashCode` are expensive operations, so we should at least minimize their usage.